### PR TITLE
Make sure values defined in schema.hrd are not lowered

### PR DESCRIPTION
**Pull request by rkhamis**
Default values were always lowered, which is very troublesome

### CODE REVIEW

File: lib/JumpScale/data/hrd/HRDSchema.py [hunk 1]
```diff
@@ -177,16 +177,16 @@ def process(self, content):
 
             hrdtype.typeclass = j.data.types.getTypeClass(ttype)
 
-            if tags.labelExists("list"):
+            if "list" in tags_dict:
```
Comment (grimpy) [inline lib/JumpScale/data/hrd/HRDSchema.py]: labels are not in this dict

File: lib/JumpScale/data/hrd/HRDSchema.py [hunk 2]
```diff
@@ -215,38 +215,38 @@ def process(self, content):
...
-                c = tags.tagGet("parent")
+            if "parent" in tags_dict:
+                c = tags_dict.get("parent")
                 hrdtype.consume_nr_min = 1
                 hrdtype.consume_nr_max = 1
                 hrdtype.consume_link = c
                 hrdtype.parent = c
 
-            if tags.labelExists("auto"):
+            if "auto" in tags_dict:
```
Comment (grimpy) [inline lib/JumpScale/data/hrd/HRDSchema.py]: label not a tag

File: lib/JumpScale/data/hrd/HRDSchema.py [hunk 3]
```diff
@@ -215,38 +215,38 @@ def process(self, content):
...
                 hrdtype.auto = True
 
-            if tags.labelExists("optional"):
+            if "optional" in tags_dict:
```
Comment (grimpy) [inline lib/JumpScale/data/hrd/HRDSchema.py]: label not a tag

**Comment by muhamadazmy** [comment]
Won't it easier (and probably cleaner) that the tagExists method (or labelExsts) do a key.lower() itself. so you don't have to copy all the tags to a dict and then check against the dict ?

File: lib/JumpScale/data/hrd/HRDSchema.py (hunk 1 above)
Comment (rkhamis) [inline lib/JumpScale/data/hrd/HRDSchema.py]: In the dict, I see
```
tags_dict
{'auto': True, 'type': 'str'}
```
So it's in there.

**Comment by rkhamis** [comment]
@muhamadazmy Indeed. But that thing is used allover and I don't want to change its behavior

**Comment by grimpy** [comment]
what would be better is that you use propercasing everywhere

**Comment by rkhamis** [comment]
@grimpy absolutely. Now ask me if I want to break all existing actors and blueprints.

**Comment by muhamadazmy** [comment]
i think the tag key should be case insensitive, so it's okay if we apply it everywhere. but u are the owner of the product, so u decide.

Also i think u should use `str.casefold()` instead of `str.lower()` for proper case-insinsitive comparison.

**Comment by rkhamis** [comment]
@muhamadazmy first time I know about casefold. Very cool. Done!

**Pull request merged**